### PR TITLE
Restores tests to set_catkeys_and_barcodes_job_spec.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made?
Tests were removed in https://github.com/sul-dlss/argo/pull/2693/files#diff-9a7dc28d99cfc95d006d7f19b549215a08e149e9a9a050c1f05f4b6bd1019896L89-L91

## How was this change tested?
Unit

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

File: spec/jobs/set_catkeys_and_barcodes_job_spec.rb [hunk 1]
```diff
@@ -75,6 +79,10 @@
     end
 
     context 'when catkey and barcode selected' do
+      before do
+        allow(ItemChangeSet).to receive(:new).and_return(change_set1, change_set2, change_set3)
```
Comment (jmartin-sul) [inline spec/jobs/set_catkeys_and_barcodes_job_spec.rb:83]: neat, didn't know about this construct for multiple returns

**Review by jcoyne — APPROVED** [review]

File: spec/jobs/set_catkeys_and_barcodes_job_spec.rb [hunk 2]
```diff
@@ -87,11 +95,14 @@
             }
           }
         expect(subject).to receive(:with_bulk_action_log).and_yield(buffer)
-        expect(subject).to receive(:update_catkey_and_barcode).with(ItemChangeSet, buffer)
-        expect(subject).to receive(:update_catkey_and_barcode).with(ItemChangeSet, buffer)
-        expect(subject).to receive(:update_catkey_and_barcode).with(ItemChangeSet, buffer)
+        expect(subject).to receive(:update_catkey_and_barcode).with(change_set1, buffer)
+        expect(subject).not_to receive(:update_catkey_and_barcode).with(change_set2, buffer)
```
Comment (jmartin-sul) [inline spec/jobs/set_catkeys_and_barcodes_job_spec.rb:99]: so what's being tested here is how `change_setN#changed?` is acted upon, right?  as opposed to, e.g., whether the `ItemChangeSet` correctly determines if there's a change or not between `item1`'s barcode value and `barcodes[0]` differ?

**Review by jmartin-sul — COMMENTED** [review]
one small question for my own edification, since this is already merged and i'd have approved anyway

File: spec/jobs/set_catkeys_and_barcodes_job_spec.rb (hunk 2 above)
Comment (justinlittman) [reply spec/jobs/set_catkeys_and_barcodes_job_spec.rb:99]: It is primarily to test that the correct change set was constructed as that will determine how the cocina model is updated.

**Pull request merged**